android/src/main/java/com/twiliovoicereactnative/CallMessageListenerProxy.java

```diff
@@ -54,21 +63,22 @@ public void onMessageSent(String voiceEventSID) {
   }
 
   @Override
-  public void onMessageReceived(CallMessage callMessage) {
+  public void onMessageReceived(String callSid, CallMessage callMessage) {
     logger.debug("onMessageReceived");
 
-    // notify JS layer ScopeCall
-    getJSEventEmitter().sendEvent(ScopeCall,
-      constructJSMap(
-        new Pair<>(VoiceEventType, CallEventMessageReceived),
-        new Pair<>(JSEventKeyCallMessageInfo, serializeCallMessage(callMessage))
-      )
-    );
+    //final call record
+    final CallRecord callRecord =
+      Objects.requireNonNull(getCallRecordDatabase().get(new CallRecord(callSid)));
 
-    // notify JS layer ScopeCallInvite
-    getJSEventEmitter().sendEvent(ScopeCallInvite,
+    // notify JS layer ScopeCallInvite or ScopeCall
```

Comment:
could we clean this up a tad, maybe do this..
```
final String event = (CallRecord.CallInviteState.ACTIVE == callRecord.getCallInviteState() ? ScopeCallInvite : ScopeCall;
final WritableMap serializedCallMap = CallRecord.CallInviteState.ACTIVE == callRecord.getCallInviteState()) ? null : serializeCall(callRecord);
getJSEventEmitter().sendEvent(
   event, 
   constructJSMap(
      new Pair<>(VoiceEventType, CallEventMessageReceived),
      new Pair<>(JS_EVENT_KEY_CALL_INFO, serializedCallMap),
      new Pair<>(JSEventKeyCallMessageInfo, serializeCallMessage(callMessage))
    )
);    
```

Reply:
i like it 👍